tests/accessor/accessor_common.h

```diff
@@ -0,0 +1,451 @@
+
+          auto acc = get_accessor_functor(data_buf, cgh);
+          if constexpr (Target == sycl::target::host_task) {
+            cgh.host_task([=] {
+              read_write_zero_dim_acc<DataT, AccessMode>(acc, res_acc);
+            });
+          } else if constexpr (Target == sycl::target::device) {
+            cgh.parallel_for_work_group(r, [=](sycl::group<1>) {
+              read_write_zero_dim_acc<DataT, AccessMode>(acc, res_acc);
+            });
+          }
+        })
+        .wait_and_throw();
+  }
+
+  if constexpr (AccessMode != sycl::access_mode::write) {
+    CHECK(compare_res);
+  }
+  if constexpr (AccessMode != sycl::access_mode::read) {
+    CHECK(value_helper::are_equal(some_data, changed_val));
+  }
+}
+
+/**
+ * @brief Function that tries to read or/and write depending on AccessMode
+ * parameter. Results of compare will be stored in res_acc
+ *
+ * @tparam DataT Type of underlying data
+ * @tparam Dimension Dimensions of the accessor
+ * @tparam AccessMode Access mode of the accessor
+ * @tparam AccT Type of testing accessor
+ * @tparam ResultAccT Type of accessor for storing result
+ * @param testing_acc Instance of sycl::accessor to read/write
+ * @param res_acc Accessor for storing result
+ */
+template <typename DataT, int Dimension, sycl::access_mode AccessMode,
+          typename AccT, typename ResultAccT>
+void read_write_acc(AccT testing_acc, ResultAccT res_acc) {
+  DataT other_data(expected_val);
+  auto id = util::get_cts_object::id<Dimension>::get(0, 0, 0);
+
+  if constexpr (AccessMode != sycl::access_mode::write) {
+    res_acc[0] = value_helper::are_equal(testing_acc[id], other_data);
+  }
+  if constexpr (AccessMode != sycl::access_mode::read) {
+    value_helper::change_val(testing_acc[id], changed_val);
+  }
+}
+
+/**
+ * @brief Function helps to check common constructor of accessor
+ *
+ * @tparam AccType Type of the accessor
+ * @tparam DataT Type of underlying data
+ * @tparam Dimension Dimensions of the accessor
+ * @tparam AccessMode Access mode of the accessor
+ * @tparam Target Target of accessor
+ * @tparam GetAccFunctorT Type of functor for accessor creation
+ * @param r Range for accessors buffer
+ */
+template <accessor_type AccType, typename DataT, int Dimension,
+          sycl::access_mode AccessMode, sycl::target Target,
+          typename GetAccFunctorT>
+void check_common_constructor(GetAccFunctorT get_accessor_functor,
+                              const sycl::range<Dimension> r) {
+  auto queue = util::get_cts_object::queue();
+  bool compare_res = false;
+  DataT some_data(expected_val);
+  {
+    sycl::buffer res_buf(&compare_res, sycl::range(1));
+    sycl::buffer<DataT, Dimension> data_buf(&some_data, r);
+
+    queue
+        .submit([&](sycl::handler& cgh) {
+          sycl::accessor res_acc(res_buf);
+          auto acc = get_accessor_functor(data_buf, cgh);
+
+          if (acc.is_placeholder()) {
+            cgh.require(acc);
+          }
```

Comment:
Upd: found `AccType` dependency in #325 : https://github.com/KhronosGroup/SYCL-CTS/pull/325/files#diff-8067235ffc00be4a68e8da8448acbf789b606099c5dbc1e969df066646bd2f64R410.

Reply:
> For me, the link goes just to the Files changed tab not to a concrete place.

Yep, looks like github restriction. It doesn't show exact line in `accessor_common.h` until the diff is loaded by clicking on "load diff".

> I stood for waiting until all PRs are merged, not only this one.

Makes sense. Unfortunatelly looks like I'll be a bit blocked until then, but I'll try to find a workaround.

> There is a difference in API. Zero dimension accessors behave like a scalar variable without a subscript operator, unlike 1-2-3 dimension accessors([spec](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_interface_for_buffer_command_accessors) lines 155 and 158). That's why read_write_zero_dim_acc functions exist, as they change accessor without invoking the subscript operator. No doubt we can constantly check if we working with a zero dimension accessor, to separate usage of 1-2-3 dimension accessor from 0 dimension accessor, but IMO it will be too many constexpr checks, that can be avoided by a different function.

Look, I totally agree that havin a separate function within `accessor_common.h` might be really nice.
But as I see it, there is no reason to present a separate API to the user of `accessor_common.h`, as it would require to use `if constexpr (Dimension == 0)` even if there is no difference at all from the point of the caller. Looks like even `sycl::range` usage might be resolved in a generic way.
Still, moving refactoring out of this PR means this question is out of scope of this PR as well.